library/std/src/sys/windows/mod.rs

```diff
@@ -65,7 +65,8 @@ pub unsafe fn cleanup() {
 pub fn decode_error_kind(errno: i32) -> ErrorKind {
     use ErrorKind::*;
 
-    match errno as c::DWORD {
+    // The assignment to () prevents accidentally writing an entry with a missing `return`
+    let () = match errno as c::DWORD {
```

Comment:
Nice idea!

Reply:
Why not merge the two `match` statements and then you don't need `return` at all, though:?

Reply:
The `WSA` error numbers have a different type, it seems.  So the first `match` has an `as` clause.